--- a/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
+++ b/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
@@ -1779,6 +1779,6 @@
          uint32_t fn_stride =
             dev->draw_ctx->fn_set_fbds_provoking_vertex_stride;
-         uint32_t fn_addr =
+         uint64_t fn_addr =
             dev->draw_ctx->fns_bo->addr.dev + fn_idx * fn_stride;
          cs_move64_to(b, addr_reg, fn_addr);
          cs_move32_to(b, length_reg, fn_stride);
